File: nasac2018_VD/testset_for_java/testset2_mini_java/src/testcases/testset2/s02/testset2_72b.java
package testcases.testset2.s02;
import testcasesupport.*;
import java.util.Vector;

import javax.servlet.http.*;

public class testset2_72b
{
    public void foo_barSink(Vector<String> dataVector , HttpServletRequest request, HttpServletResponse response) throws Throwable
    {
        String data = dataVector.remove(2);

        if (data != null)
        {
            
            response.getWriter().println("<br>foo_bar(): data = " + data);
        }

    }

    
    public void cwe_fooSink(Vector<String> dataVector , HttpServletRequest request, HttpServletResponse response) throws Throwable
    {
        String data = dataVector.remove(2);

        if (data != null)
        {
            
            response.getWriter().println("<br>foo_bar(): data = " + data);
        }

    }
}

File: nasac2018_VD/testset_for_cc++_mini/testset2_mini/testset2_72b.cpp

#include "std_testcase.h"
#include <vector>

#include <wchar.h>

using namespace std;

namespace testset2_72
{



void foo_barSink(vector<char *> dataVector)
{
    
    char * data = dataVector[2];
    
    free(data);
}






void cwe_fooSink(vector<char *> dataVector)
{
    char * data = dataVector[2];
    
    free(data);
}


void cwe_barSink(vector<char *> dataVector)
{
    char * data = dataVector[2];
    
    
    ; 
}



} 
